src/AutorestSwift/Models/Complex/Request.swift

```diff
@@ -54,24 +54,40 @@ class Request: Metadata {
     }
 
     /// Lookup a parameter by name.
-    func parameter(for name: String) -> Parameter? {
-        return parameters?.first { $0.serializedName == name }
+    func parameter(for name: String) -> ParameterType? {
+        return parameters?.first(named: name)
     }
 }
 
 extension Request {
     /// Retrieves a single body-encoded parameter, if there is one. Fails if there is more than one.
-    var bodyParam: Parameter? {
-        var bodyParams = [Parameter]()
+    var bodyParam: ParameterType? {
+        // using a dictionary allows us to eliminate duplicates in a rudimentary fashion. A Set<T> would be
+        // better, but requires model class to conform to Hashable
```

Comment:
While this is true, both Dictionary and Set have the side effect of turning an ordered list into an unordered one. I am currently working on a fix for this, as it means the generated code is not deterministic.